Formula/arangodb.rb

```diff
@@ -26,66 +29,80 @@ class Arangodb < Formula
     cause "Fails with compile errors"
   end
 
+  resource "arangodb2" do
+    url "https://www.arangodb.com/repositories/Source/ArangoDB-2.8.10.tar.gz"
+    sha256 "3f2adba7bb4fe5a93959c6929b984b053e66d3653144ed6bb1cdedd017c65050"
+  end
+
+  resource "upgrade" do
+    url "https://www.arangodb.com/repositories/Source/upgrade3-1.0.0.tar.gz"
+    sha256 "965f899685e420530bb3c68ada903c815ebd0aa55e477d6949abba9506574011"
+  end
+
   def install
-    ENV.libcxx
+    ENV.cxx11
 
-    if build.head?
-      mkdir "arangodb-build" do
-        system "cmake", "..", "-DHOMEBREW=On", "-DUSE_OPTIMIZE_FOR_ARCHITECTURE=Off", "-DASM_OPTIMIZATIONS=Off", "-DETCDIR=#{prefix}/etc", "-DVARDIR=#{var}", *std_cmake_args
-        system "make", "install"
-      end
+    (libexec/"arangodb2/bin").install resource("upgrade")
 
-    else
-      # clang on 10.8 will still try to build against libstdc++,
-      # which fails because it doesn't have the C++0x features
-      # arangodb requires.
+    resource("arangodb2").stage do
       args = %W[
         --disable-dependency-tracking
-        --prefix=#{prefix}
+        --prefix=#{libexec}/arangodb2
         --disable-relative
-        --datadir=#{share}
         --localstatedir=#{var}
+        --program-suffix=-2.8
       ]
 
-      args << "--program-suffix=-unstable" if build.head?
-
-      if ENV.compiler != :clang
-        ENV.append "LDFLAGS", "-static-libgcc -static-libstdc++"
+      if ENV.compiler == "gcc-6"
+        ENV.append "CXXFLAGS", "-O2 -g -fno-delete-null-pointer-checks"
+        inreplace "3rdParty/Makefile.v8", "CXXFLAGS=\"", "CXXFLAGS=\"-fno-delete-null-pointer-checks "
       end
 
       system "./configure", *args
       system "make", "install"
     end
-  end
 
-  # moving the "if" inside post_install does not work
-  if build.head?
-    def post_install
-      (var/"lib/arangodb3").mkpath
-      (var/"log/arangodb3").mkpath
-    end
-  else
-    def post_install
-      (var/"arangodb").mkpath
-      (var/"log/arangodb").mkpath
+    mkdir "build" do
+      args = std_cmake_args + %W[
+        -DHOMEBREW=ON
+        -DUSE_OPTIMIZE_FOR_ARCHITECTURE=OFF
+        -DASM_OPTIMIZATIONS=OFF
+        -DETCDIR=#{etc}
+        -DVARDIR=#{var}
+      ]
+
+      if ENV.compiler == "gcc-6"
+        ENV.append "V8_CXXFLAGS", "-O3 -g -fno-delete-null-pointer-checks"
+      end
 
-      system "#{sbin}/arangod", "--upgrade"
+      system "cmake", "..", *args
+      system "make", "V=1", "Verbose=1", "VERBOSE=1", "install"
     end
   end
 
+  def post_install
+    (var/"lib/arangodb3").mkpath
+    (var/"log/arangodb3").mkpath
+
+    args = %W[
+      #{libexec}/arangodb2
+      #{var}/lib/arangodb
+      #{opt_prefix}/arangodb
```

Comment:
Fixed


Am 10.07.16 um 11:54 schrieb ilovezfs:
>
> In Formula/arangodb.rb 
> <https://github.com/Homebrew/homebrew-core/pull/2576#discussion_r70179119>:
>
> >      end
> >    end
> >
> > +  def post_install
> > +    (var/"lib/arangodb3").mkpath
> > +    (var/"log/arangodb3").mkpath
> > +
> > +    args = %W[
> > +      #{libexec}/arangodb2
> > +      #{var}/lib/arangodb
> > +      #{opt_prefix}/arangodb
>
> This is the cause of the CI failure. It should be |#{opt_prefix}| not 
> |#{opt_prefix}/arangodb|, which is redundant since |opt_prefix| means 
> |/usr/local/opt/arangodb|:
>
> |Josephs-MacBook-Pro:~ joe$ brew ruby -e "puts 
> Formula[\"arangodb\"].opt_prefix" /usr/local/opt/arangodb |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Homebrew/homebrew-core/pull/2576/files/52b71e2cc37238e9a2a9b2247a71c393e2bea13d#r70179119>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AAX7Rc-ixc5RGepNKw9B55gmc2WDyRsXks5qUMFygaJpZM4JDrTm>.
>